java/vector/src/main/java/org/apache/arrow/vector/dictionary/Dictionary.java

```diff
@@ -63,11 +64,40 @@ public boolean equals(Object o) {
       return false;
     }
     Dictionary that = (Dictionary) o;
-    return Objects.equals(encoding, that.encoding) && Objects.equals(dictionary, that.dictionary);
+    return Objects.equals(encoding, that.encoding) && equals(dictionary, that.dictionary);
   }
 
   @Override
   public int hashCode() {
     return Objects.hash(encoding, dictionary);
   }
+
+  private boolean equals(FieldVector vector1, FieldVector vector2) {
```

Comment:
This is mostly implemented in vector.util.Validator, which is used in Integration tests. It should probably be refactored, but it is ok to just call the functions for now, e.g. `compareFieldVectors`.